api/DeviceAPI.cs

```diff
@@ -34,18 +35,41 @@ public List<string> enumerateDevices(UInt64 house_id)
 	{
 		//TODO: Verify the input parameters are sufficient
 		//TODO: Implement this function
-		if (house_id == 0)
-	        {
-	            return null;
-	        }
-	        //get device list
-	        var devicelist = new List<string>();
-	        var devices = new List<Device>();
-	        var dlist =
-	            (from device in devices
-	             where device.ID.HouseID == house_id
-	             select device);
-	        return devicelist;
+        if (house_id == 0)
+        {
+            return null;
+        }
+        //get device list
+		//Interfaces(address);
+		// get some device list
+		/*
+		List<Device> fake_Devices = new List<Device>()
+		{
+			new AlarmSystem(null, null, null),
+			new CeilingFan(null, null, null),
+			new GarageDoor(null, null, null),
+			new LightSwitch(null, null, null),
+			new Thermostat(null, null, null)
+		};
+		foreach (var device in devices)
+		{
+			device.ID.HouseID = 1;
+		}
+		*/
+        var devicelist = new List<string>();
+		foreach(var dev in devices)//fake_Devices)
+		{
+			if(house_id == dev.ID.HouseID)
+				devicelist.Add(dev.Name);
```

Comment:
The string should contain everything needed to communicate with the device (URL, ID, etc,) since this string is passed to the Server in the registration step for storage and future usage.